File: backend/auth-node/src/controllers/authController.js
import { AuthService } from '../services/authService.js';
import { JwtService } from '../services/jwtService.js';
import { User } from '../models/User.js';

export class AuthController {
  static async register(req, res) {
    try {
      console.log('Register request body:', req.body);
      const response = await AuthService.registerUser(req.body);
      return res.status(200).json(response);
    } catch (error) {
      console.error('Registration error:', error);
      return res.status(400).json({
        status: 'error',
        message: error.message || 'Ошибка при регистрации'
      });
    }
  }

  static async login(req, res) {
    try {
      const { username, password } = req.body;

      const user = await User.findByUsername(username);
      if (!user) {
        return res.status(401).json({
          error: 'Неверное имя пользователя или пароль'
        });
      }

      const result = await AuthService.loginUser({ username, password });

      return res.status(200).json({
        message: result.message,
        token: result.token,
        username: result.username
      });
    } catch (error) {
      if (error.message === 'Пользователь не найден' || error.message === 'Неверный пароль') {
        return res.status(401).json({
          error: 'Неверное имя пользователя или пароль'
        });
      }
      console.error('Ошибка при входе:', error);
      return res.status(500).json({
        error: 'Произошла ошибка при входе: ' + error.message
      });
    }
  }

  static async getCurrentUser(req, res) {
    try {
      if (!req.userDetails) {
        return res.status(401).json({
          status: 'error',
          message: 'Не авторизован'
        });
      }

      return res.status(200).json({
        status: 'success',
        message: 'Данные пользователя получены',
        username: req.userDetails.username,
        roles: req.userDetails.authorities,
        timestamp: new Date().toISOString()
      });
    } catch (error) {
      return res.status(500).json({
        status: 'error',
        message: 'Ошибка при получении данных пользователя'
      });
    }
  }

  static async logout(req, res) {
    return res.status(200).json({
      status: 'success',
      message: 'Выход выполнен успешно'
    });
  }
}

File: $RECYCLE.BIN/S-1-5-21-468433717-555027351-2740982991-1002/$RQXU4IE/src/main/java/com/_x/TestRegister1/Controller/AuthController.java
package com._x.TestRegister1.Controller;

import com._x.TestRegister1.DTO.LoginRequest;
import com._x.TestRegister1.DTO.RegisterRequest;
import com._x.TestRegister1.DTO.ResponseDto;
import com._x.TestRegister1.Jwt.JwtService;
import com._x.TestRegister1.Service.AuthService;
import com._x.TestRegister1.Service.CustomUserDetailsService;
import jakarta.validation.Valid;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.BadCredentialsException;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.web.bind.annotation.*;

import java.time.LocalDateTime;
import java.util.HashMap;
import java.util.Map;
import java.util.stream.Collectors;

@RestController
@RequestMapping("/api/auth")
public class AuthController {

    private final AuthService authService;
    private final JwtService jwtService;
    private final AuthenticationManager authenticationManager;
    private final CustomUserDetailsService userDetailsService;

    public AuthController(AuthService authService, JwtService jwtService, AuthenticationManager authenticationManager, CustomUserDetailsService userDetailsService) {
        this.authService = authService;
        this.jwtService = jwtService;
        this.authenticationManager = authenticationManager;
        this.userDetailsService = userDetailsService;
    }


    @PostMapping("/register")
    public ResponseEntity<?> register(
            @Valid @RequestBody RegisterRequest request) {
        try {
            ResponseDto response = authService.registerUser(request);
            return ResponseEntity.ok(response);
        } catch (RuntimeException e) {
            return ResponseEntity
                    .badRequest()
                    .body(new ResponseDto("error", e.getMessage()));
        }
    }

    @PostMapping("/login")
    public ResponseEntity<?> login(@Valid @RequestBody LoginRequest request) {
        try {
            authenticationManager.authenticate(
                    new UsernamePasswordAuthenticationToken(
                            request.getUsername(),
                            request.getPassword()
                    )
            );

            UserDetails userDetails = userDetailsService.loadUserByUsername(request.getUsername());

            String token = jwtService.generateToken(userDetails);

            Map<String, Object> loginResponse = new HashMap<>();
            loginResponse.put("message", "Вход выполнен успешно");
            loginResponse.put("token", token);
            loginResponse.put("username", userDetails.getUsername());

            return ResponseEntity.ok(loginResponse);

        } catch (BadCredentialsException e) {
            return ResponseEntity
                    .status(HttpStatus.UNAUTHORIZED)
                    .body(Map.of("error", "Неверное имя пользователя или пароль"));
        } catch (Exception e) {
            e.printStackTrace();
            return ResponseEntity
                    .status(HttpStatus.INTERNAL_SERVER_ERROR)
                    .body(Map.of("error", "Произошла ошибка при входе: " + e.getMessage()));
        }
    }

    @GetMapping("/user")
    public ResponseEntity<?> getCurrentUser(Authentication authentication) {
        if (authentication == null || !authentication.isAuthenticated()) {
            return ResponseEntity
                    .status(HttpStatus.UNAUTHORIZED)
                    .body(new ResponseDto("error", "Не авторизован"));
        }

        UserDetails userDetails = (UserDetails) authentication.getPrincipal();

        Map<String, Object> userResponse = new HashMap<>();
        userResponse.put("status", "success");
        userResponse.put("message", "Данные пользователя получены");
        userResponse.put("username", userDetails.getUsername());
        userResponse.put("roles", userDetails.getAuthorities().stream()
                .map(GrantedAuthority::getAuthority)
                .collect(Collectors.toList()));
        userResponse.put("timestamp", LocalDateTime.now());

        return ResponseEntity.ok(userResponse);
    }

    @PostMapping("/logout")
    public ResponseEntity<?> logout() {
        return ResponseEntity.ok(new ResponseDto(
                "success",
                "Выход выполнен успешно"
        ));
    }
}